cpp/src/parquet/arrow/reader.cc

```diff
@@ -338,22 +348,37 @@ class RowGroupRecordBatchReader : public ::arrow::RecordBatchReader {
     // TODO (hatemhelal): Consider refactoring this to share logic with ReadTable as this
     // does not currently honor the use_threads option.
     std::vector<std::shared_ptr<ChunkedArray>> columns(field_readers_.size());
-    for (size_t i = 0; i < field_readers_.size(); ++i) {
-      RETURN_NOT_OK(field_readers_[i]->NextBatch(batch_size_, &columns[i]));
-      if (columns[i]->num_chunks() > 1) {
-        return Status::NotImplemented("This class cannot yet iterate chunked arrays");
+    int64_t num_rows = -1;
+
+    if (columns.empty()) {
+      num_rows = std::min(batch_size_, *row_group_remaining_size_);
```

Comment:
might be good to add a comment here why this is being done

Reply:
```suggestion
      // num_rows cannot be derived from field_readers_ so compute it using
      // row group sizes cached from metadata
      num_rows = std::min(batch_size_, *row_group_remaining_size_);
```